controllers/ContentExportImport.js

```diff
@@ -3,10 +3,40 @@
 const PLUGIN_ID = 'content-export-import';
 
 const validator = require('./validations');
+const csvHelpers = require("./csvHelpers");
 
 module.exports = {
   importContent: async (ctx) => {
     const importService = strapi.plugins[PLUGIN_ID].services['contentexportimport'];
+
+    let body = ctx.request.body;
+
+    try {
+      // Check if request has files
+      if (ctx.request.files) {
+        // Get files['source']
+        const sourceFile = ctx.request.files["source"].path;
+        const type = ctx.request.files["source"].type;
+
+        // Read and parse file['source']
+        const source = csvHelpers.readFile(
+          sourceFile,
+          type === "text/csv" ? csvHelpers.csvParser : JSON.parse
+        );
+
+        body = {
+          ...body,
+          source,
+        };
+
+        ctx.request.body = body;
```

Comment:
Can we try not to re-assign an object?
Maybe pass the whole context `ctx` to the service is not a good idea in the beginning, need to refactor the parameters passing to the service, then you can avoid reassigning the `body`.